File: question/leetcode/leetcode0001.Two-Sum/leetcode0001.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}

        for i, v in enumerate(nums):
            complement = target - v
            if complement in table:
                return [table[complement], i]
            table[v] = i

        return []

File: question/leetcode/leetcode0001.Two-Sum/leetcode0001.cs
using System.Collections.Generic;

public class Solution
{
	public int[] TwoSum(int[] nums, int target)
	{
		Dictionary<int, int> table = new Dictionary<int, int>();
		for (int i = 0; i < nums.Length; i++)
		{
			int complement = target - nums[i];
			if (table.ContainsKey(complement))
			{
				return new int[] { table[complement], i };

			}
			table[nums[i]] = i;

		}
		return new int[0];

	}
}
